### ISSUE TITLE
Npcap: cannot send through loopback interface

### ISSUE THREAD

**Issue by gpotter2** [nmap/nmap#899]
Hello,
I'm using a npcap python implementation (scapy), and i'm trying to make it fully supports the npcap loopback interface.

With latest npcap 0.90 (i also tried with 0.86, and got the same results) here is what i get:

| Interface | Was able to sniff | Was able to send a packet |
| --- | --- | --- |
| Normal interface    | Yes                 | Yes                       |
| Loopback interface | Yes                 | No                       |

The packet i'm sending is a loopback ping packet.

If i perform `ping -4 localhost` in the cmd, i correctly see the packet with wireshark/scapy on the LOOPBACK interface.

Any ideas ?

**Comment by hsluoyz** [comment]
Hi @gpotter2 ,

The ``ping`` shipped with Windows doesn't use Npcap to send packets. So we don't need to talk about it.

I have just tested with Npcap 0.90 with Nping 0.7.40 (shipped with latest Nmap 7.40).

```bash
C:\Program Files (x86)\Nmap>nping 127.0.0.1

Starting Nping 0.7.40 ( https://nmap.org/nping ) at 2017-06-01 09:20 China Standard Time
SENT (0.2970s) ICMP [127.0.0.1 > 127.0.0.1 Echo request (type=8/code=0) id=26478 seq=1] IP [ttl=64 id=5584 iplen=28 ]
RCVD (0.4380s) ICMP [127.0.0.1 > 127.0.0.1 Echo reply (type=0/code=0) id=26478 seq=1] IP [ttl=128 id=20412 iplen=28 ]
SENT (1.4530s) ICMP [127.0.0.1 > 127.0.0.1 Echo request (type=8/code=0) id=26478 seq=2] IP [ttl=64 id=5584 iplen=28 ]
RCVD (1.4530s) ICMP [127.0.0.1 > 127.0.0.1 Echo reply (type=0/code=0) id=26478 seq=2] IP [ttl=128 id=20415 iplen=28 ]
SENT (2.4540s) ICMP [127.0.0.1 > 127.0.0.1 Echo request (type=8/code=0) id=26478 seq=3] IP [ttl=64 id=5584 iplen=28 ]
RCVD (2.4540s) ICMP [127.0.0.1 > 127.0.0.1 Echo reply (type=0/code=0) id=26478 seq=3] IP [ttl=128 id=20418 iplen=28 ]
SENT (3.4690s) ICMP [127.0.0.1 > 127.0.0.1 Echo request (type=8/code=0) id=26478 seq=4] IP [ttl=64 id=5584 iplen=28 ]
RCVD (3.4690s) ICMP [127.0.0.1 > 127.0.0.1 Echo reply (type=0/code=0) id=26478 seq=4] IP [ttl=128 id=20421 iplen=28 ]
SENT (4.4690s) ICMP [127.0.0.1 > 127.0.0.1 Echo request (type=8/code=0) id=26478 seq=5] IP [ttl=64 id=5584 iplen=28 ]
RCVD (4.4690s) ICMP [127.0.0.1 > 127.0.0.1 Echo reply (type=0/code=0) id=26478 seq=5] IP [ttl=128 id=20424 iplen=28 ]

Max rtt: 0.000ms | Min rtt: 0.000ms | Avg rtt: 0.000ms
Raw packets sent: 5 (210B) | Rcvd: 5 (140B) | Lost: 0 (0.00%)
Nping done: 1 IP address pinged in 4.47 seconds

C:\Program Files (x86)\Nmap>
```

Nping is the simplest utility that uses Npcap to send and receive packets. And it works to ping localhost. So I'm afraid it's not a Npcap issue.

I think Nping can serve as a good example for how to send packets with Npcap, please refer to its code here: https://github.com/nmap/nmap/tree/master/nping

**Comment by gpotter2** [comment]
Sounds fair
